src/simulations/step.jl

```diff
@@ -71,11 +60,40 @@ function CommonSolve.step!(model::EventQueueABM)
     return model
 end
 
+function step_ahead!(queue, model_t, t::Real, model::EventQueueABM)
+    stop_time = model_t[] + t
+    while until(model_t[], stop_time, model)
+        one_step!(queue, model_t, stop_time, model)
+    end
+    return 
+end
+function step_ahead!(queue, model_t, t::Function, model::EventQueueABM)
+    while until(model_t[], t, model)
+        one_step!(queue, model_t, model)
+    end
+    return 
+end
+
+function one_step!(queue, model_t, stop_time, model)
+    if isempty(queue)
+        model_t[] = stop_time
+        return
+    end
+    event_tuple, t_event = dequeue_pair!(queue)
+    if t_event > stop_time
+        enqueue!(queue, event_tuple => t_event)
+        model_t[] = stop_time
+    else
+        process_event!(event_tuple, model)
+        model_t[] = t_event
```

Comment:
I think here we first need to update the model time and then process the event. Because, the event occurs at the even time. So that if the event reads the model time, the correct time to read is the time the event is supposed to occur at.

Reply:
right :+1: